rcl/include/rcl/time.h

```diff
@@ -123,6 +141,102 @@ RCL_WARN_UNUSED
+ * On Windows this is lock-free if the C11's stdatomic.h function
+ * atomic_is_lock_free() returns true for atomic_int_least64_t.
+ *
+ * \param[out] now a struct in which the current time is stored
+ * \return RCL_RET_OK if the current time was successfully obtained, or
+ *         RCL_RET_INVALID_ARGUMENT if any arguments are invalid, or
+ *         RCL_RET_ERROR an unspecified error occur.
+ */
+RCL_PUBLIC
+RCL_WARN_UNUSED
+rcl_ret_t
+rcl_ros_time_point_now(rcl_ros_time_point_t * now);
+
+
+/// Set the current time as a rcl_ros_time_point_t struct.
+/* This function sets the time for the simulation_time. If unset it will set
+ * simulation_time active.
+ *
+ * This function is intended only to be used by rcl internals to set
+ * the simulated time.
+ *
+ * The resolution is nanoseconds.
+ *
+ * The now argument must point to an allocated rcl_ros_time_point_t struct,
+ * as the value is copied out of this variable.
+ *
+ * This function may allocate heap memory when an error occurs.
+ * This function is thread-safe.
+ * This function is lock-free, with an exception on Windows.
+ * On Windows this is lock-free if the C11's stdatomic.h function
+ * atomic_is_lock_free() returns true for atomic_int_least64_t.
+ *
+ * \param[in] now a struct containing the current time to be stored
+ * \return RCL_RET_OK if the current time was successfully obtained, or
+ *         RCL_RET_INVALID_ARGUMENT if any arguments are invalid, or
+ *         RCL_RET_ERROR an unspecified error occur.
+ */
+
+RCL_PUBLIC
+RCL_WARN_UNUSED
+rcl_ret_t
+rcl_set_ros_time_point_now(rcl_ros_time_point_t * now);
+
+
+/// Disable ros time override to simulation time.
+/* This function turns off the simulation time override.
+ * When simulation time is enabled ros time will reflect
+ * the simulated time.
+ * \return RCL_RET_OK if the current time was successfully obtained, or
+ *         RCL_RET_ERROR an unspecified error occur.
+ */
+RCL_PUBLIC
+RCL_WARN_UNUSED
+rcl_ret_t
+rcl_disable_ros_time_point(void);
+
+/// Enable ros time override to simulation time.
+/* This function turns on the simulation time override.
+ * When simulation time is enabled ros time will reflect
+ * the simulated time.
+ * \return RCL_RET_OK if the current time was successfully obtained, or
+ *         RCL_RET_ERROR an unspecified error occur.
+ */
+RCL_PUBLIC
+RCL_WARN_UNUSED
+rcl_ret_t
+rcl_enable_ros_time_point(void);
+
+
+/// Get wiether ros time override to simulation time is enabled.
+/* This function turns on the simulation time override.
+ * When simulation time is enabled ros time will reflect
+ * the simulated time.
+ * \param[out] is_enabled whether simulation time is active.
+ * \return RCL_RET_OK if the value was queried correctly.
+ *         RCL_RET_INVALID_ARGUMENT if any arguments are invalid, or
+ *         RCL_RET_ERROR an unspecified error occur.
+ */
+RCL_PUBLIC
+RCL_WARN_UNUSED
+rcl_ret_t
+rcl_is_ros_time_point_enabled(bool * is_enabled);
```

Comment:
Can this fail? If not should it just return a bool?

Is this affected by `rcl_init` and `rcl_shutdown`?